supernode/server/api/api.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+var handlerCategories map[string][]*HandlerSpec
+
+const (
+	// VersionedPrefix is recommended, any new API should start with this prefix.
+	VersionedPrefix = "/api/v1"
+
+	// LegacyPrefix is deprecated, just for compatibility with the old version.
+	LegacyPrefix = ""
+)
+
+var (
+	VersionedCategory = &Category{Name: "versioned api", Prefix: VersionedPrefix}
+	LegacyCategory    = &Category{Name: "legacy api", Prefix: LegacyPrefix}
+)
+
+// Category groups the APIs.
+type Category struct {
+	Name   string
+	Prefix string
+}
+
+// Register add an API into supernode server with current api version.
+func Register(handlers ...*HandlerSpec) {
+	for _, h := range handlers {
+		registerAPI(VersionedCategory, h)
+	}
+}
+
+// RegisterLegacy is deprecated, just for compatibility with the old version,
+// please do not use it to add new API.
+func RegisterLegacy(h *HandlerSpec) {
+	registerAPI(LegacyCategory, h)
+}
+
+func registerAPI(c *Category, h *HandlerSpec) {
```

Comment:
How about to allow register user-defined Category,  may be we want to register more than one groups of API.

Reply:
I think about it again, it's better to provide an extension category for users to register APIs. Users can groups their own customized APIs in this extension category, not in Dragonfly core APIs category.
WDYT?

Reply:
```json
GET /api/ext
```
And supernode provides the above API to list the current registered APIs.